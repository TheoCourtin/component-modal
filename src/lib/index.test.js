import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Modal from "./Modal";

describe("Composant Modal", () => {
  test("rend correctement", () => {
    const { getByTestId } = render(<Modal />);
    expect(getByTestId("modal")).toBeInTheDocument();
  });

  test("affiche le titre et le message correctement", () => {
    const title = "Exemple de titre";
    const message = "Exemple de message";
    const { getByText } = render(<Modal title={title} message={message} />);
    expect(getByText(title)).toBeInTheDocument();
    expect(getByText(message)).toBeInTheDocument();
  });

  test("appelle la fonction closeModal lors du clic sur le bouton de fermeture", () => {
    const closeModal = jest.fn();
    const { getByTestId } = render(<Modal closeModal={closeModal} />);
    const closeButton = getByTestId("close-button");
    fireEvent.click(closeButton);
    expect(closeModal).toHaveBeenCalledTimes(1);
  });
});
