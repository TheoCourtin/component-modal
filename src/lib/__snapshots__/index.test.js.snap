exports[`Modal component > renders correctly 1`] = `
<div
  className="modal"
>
  <button
    className="close-btn"
  >
    <img
      alt="Close Modal Btn"
      src="/src/lib/assets/close.svg"
    />
  </button>
  <div
    className="img-wrapper"
    style={
      {
        "display": "none",
      }
    }
  >
    <img />
  </div>
  <div
    className="content-wrapper"
  >
    <h3 />
    <p />
  </div>
</div>
`;